script/vsts/platforms/windows.yml

```diff
@@ -20,221 +20,48 @@ jobs:
-             )
-            ) ELSE (
-              ECHO Pull request build, no code signing will be performed
-              node script\vsts\windows-run.js script\build.cmd --no-bootstrap --compress-artifacts
-            )
-          )
-        env:
-          GITHUB_TOKEN: $(GITHUB_TOKEN)
-          BUILD_ARCH: $(buildArch)
-          ATOM_RELEASE_VERSION: $(ReleaseVersion)
-          ATOM_WIN_CODE_SIGNING_CERT_DOWNLOAD_URL: $(ATOM_WIN_CODE_SIGNING_CERT_DOWNLOAD_URL)
-          ATOM_WIN_CODE_SIGNING_CERT_PASSWORD: $(ATOM_WIN_CODE_SIGNING_CERT_PASSWORD)
-          IS_RELEASE_BRANCH: $(IsReleaseBranch)
-          IS_SIGNED_ZIP_BRANCH: $(IsSignedZipBranch)
-          npm_config_build_from_source: true
-        displayName: Build Atom
-
-      - script: node script\vsts\windows-run.js script\test.cmd
-        env:
-          CI: true
-          CI_PROVIDER: VSTS
-          ATOM_JASMINE_REPORTER: list
-          TEST_JUNIT_XML_ROOT: $(Common.TestResultsDirectory)\junit
-          BUILD_ARCH: $(buildArch)
-        displayName: Run tests
-        condition: and(succeeded(), ne(variables['Atom.SkipTests'], 'true'))
-
-      - script: >
-          node script\vsts\windows-run.js script\postprocess-junit-results.cmd
-          --search-folder %TEST_JUNIT_XML_ROOT% --test-results-files "**/*.xml"
-        env:
-          TEST_JUNIT_XML_ROOT: $(Common.TestResultsDirectory)\junit
-        displayName: Post-process test results
-        condition: ne(variables['Atom.SkipTests'], 'true')
-
-      - task: PublishTestResults@2
-        inputs:
-          testResultsFormat: JUnit
-          searchFolder: $(Common.TestResultsDirectory)\junit
-          testResultsFiles: '**/*.xml'
-          mergeTestResults: true
-          testRunTitle: Windows $(buildArch)
-        condition: ne(variables['Atom.SkipTests'], 'true')
-
-      - script: |
-          IF NOT EXIST "%ARTIFACT_STAGING_DIR%\crash-reports" MKDIR "%ARTIFACT_STAGING_DIR%\crash-reports"
-          IF EXIST "%Temp%\Atom Crashes" (
-            FOR %%a in ("%Temp%\Atom Crashes\*.dmp") DO XCOPY "%%a" "%ARTIFACT_STAGING_DIR%\crash-reports" /I
-          )
-        displayName: Stage crash reports
-        condition: failed()
-        env:
-          ARTIFACT_STAGING_DIR: $(Build.ArtifactStagingDirectory)
+      - template: templates/build.yml
 
-      - task: PublishBuildArtifacts@1
-        inputs:
-          PathtoPublish: $(Build.ArtifactStagingDirectory)/crash-reports
-          ArtifactName: crash-reports
-        displayName: Publish crash reports on non-release branch
-        condition: and(failed(), eq(variables['ATOM_RELEASES_S3_KEY'], ''))
+      - template: templates/test.yml
 
-      - script: >
-          node $(Build.SourcesDirectory)\script\vsts\upload-crash-reports.js --crash-report-path "%ARTIFACT_STAGING_DIR%\crash-reports" --s3-path "vsts-artifacts/%BUILD_ID%/"
+      - pwsh: |
+          if ($env:BUILD_ARCH -eq "x64") {
+            $env:FileID="-x64"
```

Comment:
"`FileID`" variable name: Can we make this more clear what it is used for? Call it "`Arch`" or "`WindowsArch`", or something similar?

Or even "`Dashx64`"?

Reply:
Again, this requires rebasing and changing the commit history. If you had given me a notice earlier we could do this.